### PR TITLE
Include flow reference when queuing messages to courier

### PR DESCRIPTION
**Pull request by rowanseymour**
and add `omitempty` to some existing fields

### CODE REVIEW

File: core/models/msgs.go [hunk 1]
```diff
@@ -122,9 +122,9 @@ type Msg struct {
 		ErrorCount           int                `db:"error_count"     json:"error_count"`
 		NextAttempt          *time.Time         `db:"next_attempt"    json:"next_attempt"`
 		FailedReason         MsgFailedReason    `db:"failed_reason"   json:"-"`
-		ExternalID           null.String        `db:"external_id"     json:"external_id"`
```
Comment (rowanseymour) [inline core/models/msgs.go:125]: a message being queued to courier never has this set

File: core/models/msgs.go [hunk 2]
```diff
@@ -116,15 +116,15 @@ type Msg struct {
 		QueuedOn             time.Time          `db:"queued_on"       json:"queued_on"`
 		Direction            MsgDirection       `db:"direction"       json:"direction"`
 		Status               MsgStatus          `db:"status"          json:"status"`
-		Visibility           MsgVisibility      `db:"visibility"      json:"visibility"`
```
Comment (rowanseymour) [inline core/models/msgs.go:119]: I'll test of course but courier shouldn't need this.. it's always 'V' for anything queued by mailroom

File: core/handlers/session_triggered_test.go [hunk 3]
```diff
@@ -44,7 +44,7 @@ func TestSessionTriggered(t *testing.T) {
 		{
 			Actions: handlers.ContactActionMap{
 				testdata.Cathy: []flows.Action{
-					actions.NewStartSession(handlers.NewActionUUID(), simpleFlow.FlowReference(), nil, []*flows.ContactReference{contactRef}, []*assets.GroupReference{groupRef}, nil, true),
+					actions.NewStartSession(handlers.NewActionUUID(), simpleFlow.Reference(), nil, []*flows.ContactReference{contactRef}, []*assets.GroupReference{groupRef}, nil, true),
```
Comment (rowanseymour) [inline core/handlers/session_triggered_test.go:47]: just renamed for consistency

File: core/models/msgs.go (hunk 1 above)
Comment (norkans7) [reply core/models/msgs.go:125]: What about those resent?

Comment (rowanseymour) [reply core/models/msgs.go:125]: if a message is failed, that would mean it wouldn't have an external id no? And if it did, resending means getting a new external id

**Review by norkans7 — APPROVED** [review]

**Review by ericnewcomer — APPROVED** [review]

**Pull request merged**